bigbluebutton-html5/imports/ui/components/presentation/component.jsx

```diff
@@ -1,78 +1,248 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+import WhiteboardOverlayContainer from '/imports/ui/components/whiteboard/whiteboard-overlay/container';
+import WhiteboardToolbarContainer from '/imports/ui/components/whiteboard/whiteboard-toolbar/container';
 import PollingContainer from '/imports/ui/components/polling/container';
+import CursorWrapperContainer from './cursor/cursor-wrapper-container/container';
 import AnnotationGroupContainer from '../whiteboard/annotation-group/container';
-import Cursor from './cursor/component';
 import PresentationToolbarContainer from './presentation-toolbar/container';
+import PresentationOverlayContainer from './presentation-overlay/container';
 import Slide from './slide/component';
 import styles from './styles.scss';
 
+
 export default class PresentationArea extends React.Component {
-  constructor(props) {
-    super(props);
+  constructor() {
+    super();
+
+    this.state = {
+      paperWidth: 0,
+      paperHeight: 0,
+      showSlide: false,
+    };
+
+    this.getSvgRef = this.getSvgRef.bind(this);
+  }
+
+  componentDidMount() {
+    // adding an event listener to scale the whiteboard on 'resize' events sent by chat/userlist etc
+    window.addEventListener('resize', () => {
+      setTimeout(this.handleResize.bind(this), 0);
+    });
+
+    const { presentationPaper, whiteboardSizeAvailable } = this;
+    this.getInitialPaperSizes(presentationPaper, whiteboardSizeAvailable);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  // returns a ref to the svg element, which is required by a WhiteboardOverlay
+  // to transform screen coordinates to svg coordinate system
+  getSvgRef() {
+    return this.svggroup;
+  }
+
```

Comment:
Both `getInitialPaperSizes` and `handleResize` do the same thing, just change the state set, shouldn't the common code be on a function?

Reply:
That's a good point.